File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
       int ans=0;
      int  low=0;
      int  high=height.size()-1;
    while(low<high){
      int res=min(height[low],height[high])*(high-low);
      ans=max(ans,res);
      if(height[low]<height[high])
      low++;
      else
      high--;
    }
    return ans;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int n = height.length;
        int r = n-1;
        int maxi = 0;
        while(l < r){
            int b = Math.min(height[l],height[r])*Math.abs(r - l);
            maxi = Math.max(maxi,b);
            if(height[l] < height[r]){
                l++;
            }
            else
            r--;

        }
        return maxi;
    }
}